src/front/wgsl/mod.rs

```diff
@@ -181,6 +262,75 @@ impl<'a> Error<'a> {
                 labels: vec![(ident_span.clone(), "unknown identifier".into())],
                 notes: vec![],
             },
+            Error::UnknownScalarType(ref bad_span) => ParseError {
+                message: format!("unknown scalar type: '{}'", &source[bad_span.clone()]),
+                labels: vec![(bad_span.clone(), "unknown scalar type".into())],
+                notes: vec!["Valid scalar types are f16, f32, f64, i8, i16, i32, i64, u8, u16, u32, u64, bool".into()],
```

Comment:
Are these supposed to be included here yet? There's listed under [Reserved Keywords](https://gpuweb.github.io/gpuweb/wgsl/#reserved-keywords) and IIRC why the width validation message only listed 32

Reply:
I suppose this is forward looking, and the use of unavailable types is caught by the validation anyway.
So we can go either way.

Reply:
I added that note because those scalar types are exactly what the lexer accepts right now:
![image](https://user-images.githubusercontent.com/1144597/120019861-3e64a300-bfe9-11eb-893c-d693e6aad9b6.png)

and as you say it will get caught in validation if the type isn't actually supported.